x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -347,19 +348,49 @@ export class KnowledgeBaseService {
     }));
   }
 
+  private async getConnectorIndices(
+    client: ElasticsearchClient,
+    uiSettingsClient: IUiSettingsClient
+  ) {
+    const customSearchConnectorIndex = await uiSettingsClient.get<string>(
+      aiAssistantSearchConnectorIndexPattern
+    );
+
+    if (customSearchConnectorIndex) {
+      return customSearchConnectorIndex.split(',');
+    }
+
+    const response = (await client.transport.request({
+      method: 'GET',
+      path: '_connector',
+    })) as { results: Array<{ index_name: string }> };
+    const connectorIndices = response.results.map((result) => result.index_name);
+
+    // preserve backwards compatibility with 8.14 (may not be needed in the future)
+    if (isEmpty(connectorIndices)) {
+      return ['search-*'];
+    }
+
+    return connectorIndices;
+  }
```

Comment:
Maybe over-optimizing, but can we do these in parallel? I assume both requests are pretty cheap, but there is some overhead and one blocking batch of requests might be better than two consecutive blocking requests

Reply:
Happy to do that but one thought:
Doing it in parallel will improve perf marginally when the user has not specified `customSearchConnectorIndex`. In the case where the user has specified `customSearchConnectorIndex` we'll block until the call to get connectors has completed, even though it's not used. I expect this to be the slowest of the two.

Reply:
Calls are executed in parallel 69baf8c78f762c7c685153453899e4238d86c125.

Reply:
> we can still return as soon as the uiSettings call is finished and we figure out that the user has configured the index

We could. I added a follow-up commit (45117d6e1c65651b1772fc0872cb07f6c043b158) to do this but not sure how well it reads. Anyway, now it's there so happy to leave it in